### ISSUE TITLE
Prevent MongoDB crashes, by reducing the number of opened cursors

### ISSUE THREAD

**Issue by adrienjoly** [openwhyd/openwhyd#318]
See https://github.com/openwhyd/openwhyd/issues/202#issuecomment-605647691. Possibly related to #144.

Initial cause hypothesis: the crash would be caused by stress brought to the database by the digest worker. (for email notifications)

In the end, it seems that making sure that we close cursors opened by `forEach2()` (cf PR #405) was key to reducing that stress:

<img width="559" alt="Capture d’écran 2020-11-23 à 13 39 20" src="https://user-images.githubusercontent.com/531781/99963377-fcd11200-2d91-11eb-8e5a-5ecaec9b2f34.png">

**Comment by adrienjoly** [comment]
This is not a proof that the notifications worker was the cause of this morning's crash, but we can see that the one at 11:23 took 40 seconds to run:

<img width="727" alt="Capture d’écran 2020-11-10 à 14 05 43" src="https://user-images.githubusercontent.com/531781/98677973-20419900-235e-11eb-9d96-70117fa8d012.png">

... and many cursors where opened at that time:

<img width="1280" alt="Capture d’écran 2020-11-10 à 12 46 24" src="https://user-images.githubusercontent.com/531781/98678105-5848dc00-235e-11eb-97d8-21a698752726.png">

... and many mongodb page faults:

<img width="1280" alt="Capture d’écran 2020-11-10 à 11 57 05" src="https://user-images.githubusercontent.com/531781/98678334-acec5700-235e-11eb-86ac-f0fb8e55b5e7.png">

Datadog notified that MongoDB stopped to respond, starting at 11:28.

**Comment by adrienjoly** [comment]
Extract from the access log:

```
Tue, 10 Nov 2020 09:56:58 GMT GET /genre/Reggae (42951ms) ua=Mozilla/5.0 
Tue, 10 Nov 2020 10:07:03 GMT GET / (857ms) ua=Mozilla/5.0 
Tue, 10 Nov 2020 10:07:03 GMT GET /post (1016ms) uid=5361647f71eaec19b57037e4 ua=Mozilla/5.0 

Tue, 10 Nov 2020 10:16:23 GMT GET / (2243ms) uid=5361647f71eaec19b57037e4 ua=Mozilla/5.0 
Tue, 10 Nov 2020 10:16:26 GMT GET /api/notif (1464ms) uid=5361647f71eaec19b57037e4 ua=Mozilla/5.0 

Tue, 10 Nov 2020 10:17:17 GMT GET /u/5361647f71eaec19b57037e4 (1023ms) uid=5361647f71eaec19b57037e4 ua=Mozilla/5.0 

Tue, 10 Nov 2020 10:23:13 GMT GET /u/5361647f71eaec19b57037e4 (536ms) uid=5361647f71eaec19b57037e4 ua=Mozilla/5.0 
Tue, 10 Nov 2020 10:23:15 GMT GET /c/57be0d1c3499710704d311b9 (1581ms) ua=Mozilla/5.0  # ℹ️ this should be faster
Tue, 10 Nov 2020 10:23:14 GMT GET /u/516891667e91c862b2ace925 (6029ms) ua=Mozilla/5.0  # ℹ️ this profile is light
Tue, 10 Nov 2020 10:23:16 GMT GET / (4348ms) ua=Mozilla/5.0 
Tue, 10 Nov 2020 10:23:19 GMT GET /u/51b0af987e91c862b2ae8e1e/playlist/2 (1185ms) ua=Mozilla/5.0 
Tue, 10 Nov 2020 10:23:28 GMT GET / (2981ms) ua=Mozilla/5.0 
Tue, 10 Nov 2020 10:23:33 GMT GET / (13050ms) ua=Mozilla/5.0 

Tue, 10 Nov 2020 10:23:29 GMT POST /api/post (46080ms) uid=5361647f71eaec19b57037e4 ua=Mozilla/5.0 
Tue, 10 Nov 2020 10:23:52 GMT GET /api/notif (29503ms) uid=553e4ed25969c78f98df65fa ua=Mozilla/5.0 
Tue, 10 Nov 2020 10:24:17 GMT GET /api/notif (63159ms) uid=518b5a447e91c862b2adea1a ua=Mozilla/5.0 
Tue, 10 Nov 2020 10:58:32 GMT GET /hot/jazz (18297ms) ua=Mozilla/5.0 
Tue, 10 Nov 2020 10:58:58 GMT GET /hot/folk (15913ms) ua=Mozilla/5.0 
```

=> Observations:

- `/genre` and `/hot` pages are very slow => we should at least get rid of `/genre`

**Comment by adrienjoly** [comment]
A lot of slowness and a `502` on openwhyd.org, since this afternoon ~4pm, and despite a restart after deploying `dd-agent` APM at 7pm CET:

<img width="1280" alt="Capture d’écran 2020-11-14 à 19 13 20" src="https://user-images.githubusercontent.com/531781/99154096-8a837200-26ad-11eb-9b02-456746c31a08.png">

**Comment by adrienjoly** [comment]
Some MongoDB stats from Datadog:

![image](https://user-images.githubusercontent.com/531781/99902523-ad6de180-2cbe-11eb-8096-1a9343cf7c2e.png)

![image](https://user-images.githubusercontent.com/531781/99902530-c1b1de80-2cbe-11eb-82a3-9ee8e8b45b99.png)

**Comment by adrienjoly** [comment]
fixed by PR #405
